kinde_sdk/auth/oauth.py

```
@@ -643,4 +664,163 @@ def get_tokens(self, user_id: str) -> Dict[str, Any]:
+        Returns:
+            bool: True if access is allowed, False otherwise
+        """
+        try:
+            try:
+                loop = asyncio.get_running_loop()
+                # If we're already in an event loop (e.g., FastAPI), instruct callers to use the async API.
+                if loop.is_running():
+                    raise RuntimeError(
+                        "check_route_access() cannot be used inside a running event loop. "
+                        "Use `await validate_route_access(...)` instead."
+                    )
```

Ruff (0.14.3)

Abstract `raise` to an inner function

(TRY301)